### PR TITLE
Minor: update ci

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Follow #7946

## What changes are included in this PR?
Update the ci for example check

## Are these changes tested?
Yes

### CODE REVIEW

**Comment by Weijun-H** [comment]
Draft until #7946 merges

**Comment by Weijun-H** [comment]
The ci doesn't run all examples, should I add the missing one?

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -136,22 +136,10 @@ jobs:
...
-          cargo run --example parquet_sql
-          cargo run --example parquet_sql_multiple_files
-          cargo run --example memtable
-          cargo run --example rewrite_expr
-          cargo run --example simple_udf
-          cargo run --example simple_udaf
+          for example in $(ls datafusion-examples/examples/*.rs); do
+            example_name=$(basename "$example" .rs)
+            cargo run --example "$example_name"
+          done
```
Comment (Weijun-H) [inline .github/workflows/rust.yml:142]: Iterate all examples

**Comment by alamb** [comment]
Something doesn't seem right with this PR -- there is no change. Maybe the changes got list in the rebase. Marking as a draft. Thank you @Weijun-H

**Comment by Weijun-H** [comment]
> Something doesn't seem right with this PR -- there is no change. Maybe the changes got list in the rebase. Marking as a draft. Thank you @Weijun-H

Hi @alamb, because in #7946 we were able to run 'cargo run --example avro_sql' without '--features=datafusion/avro', I updated the CI to ensure they are the same.

**Pull request closed without merging**